### PR TITLE
Add possibility to ignore files #571

### PR DESCRIPTION
**Pull request by spoorcc**
Implements #571

Adds ignore field:

```yaml
- name: SomeInterestingProject
  url: some-remote-server/SomeInterestingProject.git
  ignore:
  - SomeFolder/SomeOtherSubFolder
  - SomeFolder/SomeSubFolder/SomeFile.md
```

# Testing

```bash
pip install https://github.com/dfetch-org/dfetch/archive/feature/ignore-field.zip
```

### CODE REVIEW

**Comment by ben-edna** [comment]
Tests failing due to #428

File: features/fetch-with-ignore-git.feature [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            """
+            manifest:
+                version: 0.0
+                projects:
+                    - name: SomeInterestingProject
+                      url: some-remote-server/SomeInterestingProject.git
+                      src: SomeFolder/SomeSubFolder
+                      ignore:
+                         - OtherFile.txt
+                      tag: v1
```
Comment (spoorcc) [inline features/fetch-with-ignore-git.feature:26]: The repo mentioned doesn't have this tag, but it is fetched?!

**Review by spoorcc — COMMENTED** [review]
Seems something wrong in the git tag implementation, maybe due to recent change in git.py

File: features/fetch-with-ignore-git.feature (hunk 1 above)
Comment (ben-edna) [reply features/fetch-with-ignore-git.feature:26]: The generated local git repo always has an implicit v1 tag:

https://github.com/dfetch-org/dfetch/blob/c8f3b2200f5464a66dad41be79248190f191e31e/features/steps/git_steps.py#L92

**Pull request merged**